autopush/websocket.py

```diff
@@ -365,15 +368,30 @@ def _check_other_nodes(self, result):
             attrs = result.get("Attributes")
             node_id = self._get_aval(attrs.get("node_id", {}), 'S')
             uaid = self._get_aval(attrs.get("uaid", {}), 'S')
-            connected_at = self._get_aval(attrs.get("connected_at", {}), 'N')
-            url = "%s/notif/%s/%s" % (node_id, uaid, connected_at)
-            if not getattr(self, 'testing', False):
-                d = self.ap_settings.agent.request(
-                    "DELETE",
-                    url.encode("utf8"),
-                ).addCallback(self.finish_hello)
-                d.addErrback(self.log_err, extra="Failed to delete old node")
-                return
+            last_connect = self._get_aval(attrs.get("connected_at", {}), 'N')
+            existing = self.ap_settings.clients.get(uaid)
+            if existing:
+                if existing.connected_at <= last_connect:
+                    existing.sendClose()
+                if existing.connected_at > last_connect:
+                    self.sendClose()
+                    return
+            else:
```

Comment:
No 'else' here, since its possible for an existing to be there *and* the node_id is a different node. So the check here should be to do the delete if the prior node_id is not our own.